plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -7623,6 +7624,95 @@ public void testSystemTables()
         assertQueryFails("TABLE \"nation$foo\"", "\\Qline 1:1: Table '%s.%s.\"nation$foo\"' does not exist".formatted(catalog, schema));
     }
 
+    @Test
+    public void testExtraProperties()
+    {
+        String tableName = "create_table_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+        assertQuery(
+                "SELECT value FROM \"%s$properties\" where key in ('extra.property.one', 'extra.property.two')".formatted(tableName),
+                "VALUES ('one'), ('two')");
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testReplaceTableExtraProperties()
+    {
+        String tableName = "replace_table_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+        assertUpdate("CREATE OR REPLACE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.three'], ARRAY['three']))".formatted(tableName));
+
+        assertQuery(
+                "SELECT value FROM \"%s$properties\" where key in ('extra.property.three')".formatted(tableName),
+                "VALUES 'three'");
+    }
+
+    @Test
+    public void testExtraPropertiesWithCtas()
+    {
+        String tableName = "create_table_ctas_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+         assertQuery(
+                "SELECT value FROM \"%s$properties\" where key in ('extra.property.one', 'extra.property.two')".formatted(tableName),
+                "VALUES ('one'), ('two')");
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testShowCreateWithExtraProperties()
+    {
+        String tableName = format("%s.%s.show_create_table_with_extra_properties_%s", getSession().getCatalog().get(), getSession().getSchema().get(), randomNameSuffix());
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+        // Assert that SHOW CREATE TABLE does not contain extra_properties
+        assertThat((String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue())
```

Comment:
SHOW CREATE TABLE  will contain "extra_properties" now, because the table name contains this substring. :) 
Don't quite qet the `.satisfies(....)` purpose.
In any case rename the table to somethign without extra_properties to ensure that SHOW CREATE TABLE output

```
assertThat((String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue()).doesNotContain("extra_properties");
```